stored_requests/backends/db_fetcher/fetcher.go

```diff
@@ -93,6 +99,51 @@ func (fetcher *dbFetcher) FetchRequests(ctx context.Context, requestIDs []string
 	return storedRequestData, storedImpData, errs
 }
 
+func (fetcher *dbFetcher) FetchResponses(ctx context.Context, ids []string) (data map[string]json.RawMessage, errs []error) {
+	if len(ids) < 1 {
+		return nil, nil
+	}
+
+	query := fetcher.responseQueryMaker(len(ids))
+	idInterfaces := make([]interface{}, len(ids))
+	for i := 0; i < len(ids); i++ {
+		idInterfaces[i] = ids[i]
+	}
+
+	rows, err := fetcher.db.QueryContext(ctx, query, idInterfaces...)
+	if err != nil {
+		fmt.Println(err.Error(), query)
+		return nil, []error{err}
+	}
+	defer func() {
+		if err := rows.Close(); err != nil {
+			glog.Errorf("error closing DB connection: %v", err)
+		}
+	}()
+
+	storedData := make(map[string]json.RawMessage, len(ids))
+	for rows.Next() {
+		var id string
+		var data []byte
+		var dataType string
+
+		// Fixes #338
+		if err := rows.Scan(&id, &data, &dataType); err != nil {
+			return nil, []error{err}
+		}
+		storedData[id] = data
+	}
+
+	// Fixes #338
+	if rows.Err() != nil {
+		fmt.Println(rows.Err())
```

Comment:
I don't think you need to print the error to terminal.

Reply:
Right, removed